src/redux/books/books.js

```diff
@@ -1,27 +1,65 @@
-const ADD_BOOK = 'bookStore/books/ADD_BOOK';
-const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+import axios from 'axios';
+
+const GET_BOOKS_SUCCESS = 'bookStore/books/GET_BOOKS_SUCCESS';
+const GET_BOOKS_FAILURE = 'bookStore/books/GET_BOOKS_FAILURE';
+const ADD_BOOK_SUCCESS = 'bookStore/books/ADD_BOOK_SUCCESS';
+const ADD_BOOK_FAILURE = 'bookStore/books/ADD_BOOK_FAILURE';
+const REMOVE_BOOK_SUCCESS = 'bookStore/books/REMOVE_BOOK_SUCCESS';
+const REMOVE_BOOK_FAILURE = 'bookStore/books/REMOVE_BOOK_FAILURE';
 
 const initialState = [];
 
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+export const getBooksList = () => (dispatch) => axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ahmdadl563/books').then(
+  (books) => {
+    const booksList = Object.entries(books.data);
+
+    const newbooksList = [];
+
+    for (let i = 0; i < booksList.length; i += 1) {
+      const newBook = {
+        id: booksList[i][0],
+        title: booksList[i][1][0].title,
+        category: booksList[i][1][0].category,
+      };
+
+      newbooksList.push(newBook);
+    }
+    dispatch({ type: GET_BOOKS_SUCCESS, payload: newbooksList });
+  },
+  (err) => dispatch({ type: GET_BOOKS_FAILURE, err }),
+);
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+export const addBook = (book) => (dispatch) => axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ahmdadl563/books', {
+  item_id: book.id,
+  title: book.title,
+  category: book.category,
+}).then(
+  () => {
+    dispatch({ type: ADD_BOOK_SUCCESS, payload: book });
+  },
+  (err) => dispatch({ type: ADD_BOOK_FAILURE, err }),
+);
```

Comment:
- [ ] It would be better if the post actions works properly. Currently, when I try to add a book it doesn't work. Kindly fix this minor issue. 
😄 👍